packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -638,104 +636,19 @@ class FlutterPlugin implements Plugin<Project> {
-     *        directory exists, even if the surrounding flutter package does not
-     *        support the android platform (see e.g. apple_maps_flutter: 1.0.1).
-     *        So as it's included successfully it expects to be added as API.
-     *        This is only possible by taking all plugins into account, which
-     *        only appear on the `dependencyGraph` and in the `.flutter-plugins` file.
-     * So in summary the plugins are currently selected from the `dependencyGraph`
-     * and filtered then with the [doesSupportAndroidPlatform] method instead of
-     * just using the `plugins.android` list.
-     */
-    private void configureLegacyPluginEachProjects(Project project) {
-        try {
-            if (!settingsGradleFile(project).text.contains("'.flutter-plugins'")) {
-                return
-            }
-        } catch (FileNotFoundException ignored) {
-            throw new GradleException("settings.gradle/settings.gradle.kts does not exist: ${settingsGradleFile(project).absolutePath}")
-        }
-        List<Map<String, Object>> deps = getPluginDependencies(project)
-        List<String> plugins = getPluginList(project).collect { it.name as String }
-        deps.removeIf { plugins.contains(it.name) }
-        deps.each {
-            Project pluginProject = project.rootProject.findProject(":${it.name}")
-            if (pluginProject == null) {
-                // Plugin was not included in `settings.gradle`, but is listed in `.flutter-plugins`.
-                project.logger.error("Plugin project :${it.name} listed, but not found. Please fix your settings.gradle/settings.gradle.kts.")
-            } else if (doesSupportAndroidPlatform(pluginProject.projectDir.parentFile.path as String)) {
-                // Plugin has a functioning `android` folder and is included successfully, although it's not supported.
-                // It must be configured nonetheless, to not throw an "Unresolved reference" exception.
-                configurePluginProject(it)
-            /* groovylint-disable-next-line EmptyElseBlock */
-            } else {
-            // Plugin has no or an empty `android` folder. No action required.
-            }
-        }
-    }
-
-    // TODO(54566): Can remove this function and its call sites once resolved.
-    /**
-     * Returns `true` if the given path contains an `android` directory
-     * containing a `build.gradle` or `build.gradle.kts` file.
-     */
-    private Boolean doesSupportAndroidPlatform(String path) {
-        File buildGradle = new File(path, 'android' + File.separator + 'build.gradle')
-        File buildGradleKts = new File(path, 'android' + File.separator + 'build.gradle.kts')
-        if (buildGradle.exists() && buildGradleKts.exists()) {
-            logger.error(
-                "Both build.gradle and build.gradle.kts exist, so " +
-                "build.gradle.kts is ignored. This is likely a mistake."
-            )
-        }
```

Comment:
removing this also effectively reverts #140744

Reply:
I see this implementation
```groovy
    private Boolean doesSupportAndroidPlatform(String path) {
        File editableAndroidProject = new File(path, 'android' + File.separator + 'build.gradle')
        return editableAndroidProject.exists()
    }
```
below. Does replacing your implementation with this one help keep your change?

Reply:
yup, let's try that

Reply:
It looks like that PR landed after the 3.19 branch point? Keep in mind that the more complex the change is in terms of mixing pre- and post-branch code, the harder making a cherry-pick will be.

Reply:
Right ok that makes sense. This was part of https://github.com/flutter/flutter/pull/140744. The other option would be removing it. If I do that to make the cherry-pick easier, @bartekpacia will your entire PR have to be reverted?

If not, we could potentially remove it for this revert to cherry-pick and then we can land adding back in a follow-up PR so it at least lives in the master branch.

Reply:
Depending on the answer here, I'm thinking either also revert that PR or just delete this newer code for now to get this landed.

@Gustl22 could you try reading https://github.com/flutter/flutter/issues/141940#issuecomment-1912618487 and trying again? It may help you repro. I will also try to repro with/without this PR!

Reply:
@camsim99 do whatever is best for you, I can always reland my contribution later! If reverting my whole PR is easier for you then go for it.

Reply:
It's definitely easier just to delete this implementation of the method for now, so I'll go ahead with that. I will file an issue to put it back so we can track the removal, but I will try to follow this up immediately with that :)